### PR TITLE
GODRIVER-2539 Remove MinWireVersion < 6 Logic

### PR DESCRIPTION
**Pull request by prestonvasquez**
GODRIVER-2539

https://github.com/mongodb/mongo-go-driver/pull/1063 bumps the minimum wire version from 2 to 6. The scope of GODRIVER-2539 is to remove logic dependent on MinWireVersion < 6 in "x/mongo/driver/operation.go" and "mongo/change_stream.go".

### Deprecating "mongo/integration/operation_legacy_test.go"

In an internal discussion about this ticket, the following came up:

> If converting the tests to use and test OpMsg would be reasonably quick and not duplicate existing test coverage, then that could be a good way to increase existing test coverage.

The tests in this file use the function [`validateQueryWireMessage`](https://github.com/mongodb/mongo-go-driver/blob/f12df171e949167b52107d2f37ced02670408664/mongo/integration/operation_legacy_test.go#L255) which is `wiremessage.ReadQuery*` focused. Most of these "ReadQuery" functions do not have "ReadMsg" analogue, the  only one being `ReadQueryFlags`. `ReadMsgFlags` already has test coverage [here](https://github.com/mongodb/mongo-go-driver/blob/f12df171e949167b52107d2f37ced02670408664/x/mongo/driver/operation_test.go#L734). 

### Deprecating "x/mongo/driver/operation_legacy.go"

The functionality of this file is exclusively used in the two legacy branches removed from the "driver.Operation#Execute" function and was untested outside of "mongo/integration/operation_legacy_test.go".

### CODE REVIEW

File: mongo/change_stream.go [hunk 1]
```diff
@@ -287,7 +287,7 @@ func (cs *ChangeStream) executeOperation(ctx context.Context, resuming bool) err
 	// Execute the aggregate, retrying on retryable errors once (1) if retryable reads are enabled and
 	// infinitely (-1) if context is a Timeout context.
 	var retries int
-	if cs.client.retryReads && cs.wireVersion != nil && cs.wireVersion.Max >= 6 {
+	if cs.client.retryReads && cs.wireVersion != nil {
```
Comment (benjirewis) [inline mongo/change_stream.go:290]:
```suggestion
	if cs.client.retryReads {
```
The previous check seemed to set `retries = 1` if retryable reads were enabled and the wire version was known and greater than or equal to 6. I believe the `cs.wireVersion != nil` check here used the < 6 wire version logic when the wire version was unknown. I think we can remove it and always operate under the assumption that wire version is greater than 6 given that we no longer support earlier wire versions.

File: mongo/change_stream.go [hunk 2]
```diff
@@ -325,9 +325,8 @@ AggregateExecuteLoop:
 			}
 			defer conn.Close()
 
-			// If wire version is now < 6, do not retry.
 			cs.wireVersion = conn.Description().WireVersion
-			if cs.wireVersion == nil || cs.wireVersion.Max < 6 {
+			if cs.wireVersion == nil {
```
Comment (benjirewis) [inline mongo/change_stream.go:329]: See my previous comment. I think we can remove this entire wire version check.

File: x/mongo/driver/operation.go [hunk 3]
```diff
@@ -830,7 +812,7 @@ func (op Operation) retryable(desc description.Server) bool {
 			return true
 		}
 		if retryWritesSupported(desc) &&
-			desc.WireVersion != nil && desc.WireVersion.Max >= 6 &&
+			desc.WireVersion != nil &&
```
Comment (benjirewis) [inline x/mongo/driver/operation.go:815]: See my previous comment. I think we can remove the `desc.WireVersion != nil` piece of this conditional.

File: x/mongo/driver/operation.go [hunk 4]
```diff
@@ -839,7 +821,7 @@ func (op Operation) retryable(desc description.Server) bool {
 		if op.Client != nil && (op.Client.Committing || op.Client.Aborting) {
 			return true
 		}
-		if desc.WireVersion != nil && desc.WireVersion.Max >= 6 &&
+		if desc.WireVersion != nil &&
```
Comment (benjirewis) [inline x/mongo/driver/operation.go:824]: See my previous comment. I think we can remove the `desc.WireVersion != nil` piece of this conditional.

**Review by benjirewis — COMMENTED** [review]
🙌 for cleanup. Just a few comments about further simplifying wire version checks.

**Comment by benjirewis** [comment]
Now that we don't need to check wire version for resuming change streams, I wonder if we can simplify `executeOperation` in `change_stream.go` to rely on server-selection/connection-checkout in `operation.Execute` instead of [doing it manually](https://github.com/mongodb/mongo-go-driver/blob/master/mongo/change_stream.go#L241-L253). Not sure that's in the scope of this ticket, but I'll look into it + file another ticket.

File: mongo/change_stream.go [hunk 5]
```diff
@@ -325,12 +325,6 @@ AggregateExecuteLoop:
 			}
 			defer conn.Close()
 
-			// If wire version is now < 6, do not retry.
-			cs.wireVersion = conn.Description().WireVersion
```
Comment (qingyang-hu) [inline mongo/change_stream.go:329]: Do we need to update the `wireVersion`?

Comment (prestonvasquez) [reply mongo/change_stream.go:329]: Ah yes, I think we should. Nice catch, thank you!

**Review by qingyang-hu — APPROVED** [review]
LGTM!

**Review by benjirewis — APPROVED** [review]
Looks good! Thanks

**Pull request merged**